### PR TITLE
feat: allow specifying existing token secret

### PR DESCRIPTION
**Pull request by matt-march**
Adds a `mezmoApiAccessSecret` option to use instead of a manually specified access token

ref: LOG-19426

### CODE REVIEW

File: charts/edge/README.md [hunk 1]
```diff
@@ -31,7 +31,8 @@ helm install edge mezmo/edge \
 
 | **Key**                   | **Type** | **Default**     | **Description**
 | ------------------------- | -------- | --------------- | ----------------------------------------------------------------------
-| mezmoApiAccessToken       | string   |                 | Your Mezmo API access token
+| mezmoApiAccessToken       | string   |                 | Your Mezmo API access token. Mutually exclusive with `mezmoApiAccessSecret`
+| mezmoApiAccessSecret      | string   |                 | Existing secret containing your Mezmo API access token. Secret is assumed to have the token in a key named `api-access-token` and be in the same namespace as the Edge instance. Mutually exclusive with `mezmoApiAccessToken`
```
[the remark this answers is not in the document]
Comment (matt-march) [reply charts/edge/README.md]: Makes sense, updated a bit and indicated which takes precedence rather than using mutually exclusive

File: charts/edge/templates/NOTES.txt [hunk 2]
```diff
@@ -1,7 +1,8 @@
-{{- if not .Values.mezmoApiAccessToken -}}
-#####################################################################
-###  ERROR: Please specify an access token `mezmoApiAccessToken`  ###
-#####################################################################
+{{- if and (not .Values.mezmoApiAccessToken) (not .Values.mezmoApiAccessSecret) -}}
+######################################################################
+###  ERROR: Please specify an access token (`mezmoApiAccessToken`) ###
+###    or a secret containing a token (`mezmoApiAccessSecret`)     ###
+######################################################################
```
[the remark this answers is not in the document]
Comment (matt-march) [reply charts/edge/templates/NOTES.txt]: Clarified a bit to indicate which one takes precedence rather than using mutually exclusive. Also updated to block the installation instead of just showing an error

File: charts/edge/templates/NOTES.txt [hunk 3]
```diff
@@ -1,7 +1,8 @@
-{{- if not .Values.mezmoApiAccessToken -}}
-#####################################################################
-###  ERROR: Please specify an access token `mezmoApiAccessToken`  ###
-#####################################################################
+{{- if and (not .Values.mezmoApiAccessToken) (not .Values.mezmoApiAccessSecret) -}}
+######################################################################
+###  ERROR: Please specify an access token (`mezmoApiAccessToken`) ###
```
[the remark this answers is not in the document]
Comment (matt-march) [reply charts/edge/templates/NOTES.txt]: `Error` makes sense since it's functionally required to operate, though not install. Updated the flow to block installation instead of just showing the error

File: charts/edge/templates/NOTES.txt [hunk 4]
```diff
@@ -1,13 +1,3 @@
-{{- if not .Values.mezmoApiAccessToken -}}
-#####################################################################
-###  ERROR: Please specify an access token `mezmoApiAccessToken`  ###
-#####################################################################
-
-Try:
-
-  $ helm upgrade --reuse-values --set mezmoApiAccessToken=<ACCESS_TOKEN> {{ .Release.Name }} mezmo/{{ .Chart.Name }}
-
-{{ else -}}
```
Comment (matt-march) [inline charts/edge/templates/NOTES.txt:10]: Deleted this how that the check blocks installation, so this post-install message isn't shown

File: charts/edge/README.md [hunk 5]
```diff
@@ -31,7 +31,8 @@ helm install edge mezmo/edge \
 
 | **Key**                      | **Type** | **Default**      | **Description**
 | ---------------------------- | -------- | ---------------- | ----------------------------------------------------------------------
-| mezmoApiAccessToken          | string   |                  | Your Mezmo API access token
+| mezmoApiAccessToken       | string   |                 | Your Mezmo API access token. If `mezmoApiAccessSecret` is also specified it take precedence and this will be unused
```
Comment (jakedipity) [inline charts/edge/README.md:34]: nit:
```
If `mezmoApiAccessSecret` is also specified it take precedence and this will be unused
```

Besides missing an "s" after "take" and a comma or two, I find this sentence kind of hard to follow. I think changing to something like

```
This has a lower precedence than `mezmoApiAccessSecret`
```

Gets the same idea across and is much simpler to grok.

**Review by jakedipity — APPROVED** [review]
I had 1 nit comment, but it's pretty minor so approved!

File: charts/edge/README.md (hunk 5 above)
Comment (matt-march) [reply charts/edge/README.md]: That is better, thanks. Updated!

**Review by jakedipity — APPROVED** [review]
